app/src/main/java/fr/free/nrw/commons/contributions/ContributionsPresenter.java

```diff
@@ -3,104 +3,155 @@
-                Timber.d(e);
-            }
-        }
+        compositeDisposable=new CompositeDisposable();
     }
 
-    @Override
-    public void onDetachView() {
-        this.view = null;
-        if (null != cursor) {
-            try {
-                cursor.unregisterDataSetObserver(this);
-            } catch (Exception e) {//Cursor might not be already registered
-                Timber.d(e);
-            }
-        }
+    public void setLifeCycleOwner(LifecycleOwner lifeCycleOwner){
+        this.lifeCycleOwner=lifeCycleOwner;
     }
 
-    @NonNull
-    @Override
-    public Loader<Cursor> onCreateLoader(int id, @Nullable Bundle args) {
-        int preferredNumberOfUploads = repository.get(UPLOADS_SHOWING);
-        return new CursorLoader(context, BASE_URI,
-                ALL_FIELDS, "", null,
-                ContributionDao.CONTRIBUTION_SORT + "LIMIT "
-                        + (preferredNumberOfUploads>0?preferredNumberOfUploads:100));
+    public void fetchContributions() {
+        Timber.d("fetch  Contributions");
+        LiveData<List<Contribution>> liveDataContributions = repository.fetchContributions();
+        if(null!=lifeCycleOwner) {
+             liveDataContributions.observe(lifeCycleOwner, this::showContributions);
+        }
+
+        if (NetworkUtils.isInternetConnectionEstablished(CommonsApplication.getInstance()) && shouldFetchContributions()) {
+            Timber.d("fetching contributions: ");
+            view.showProgress(true);
+            this.user = sessionManager.getUserName();
+            view.showContributions(null);
+            compositeDisposable.add(userClient.logEvents(user)
+                    .subscribeOn(ioThreadScheduler)
+                    .observeOn(mainThreadScheduler)
+                    .doOnNext(mwQueryLogEvent -> Timber.d("Received image %s", mwQueryLogEvent.title()))
+                    .filter(mwQueryLogEvent -> !mwQueryLogEvent.isDeleted()).doOnNext(mwQueryLogEvent -> Timber.d("Image %s passed filters", mwQueryLogEvent.title()))
+                    .map(image -> {
+                        Contribution contribution = new Contribution(null, null, image.title(),
+                                "", -1, image.date(), image.date(), user,
+                                "", "", STATE_COMPLETED);
+                        return contribution;
+                    })
+                    .toList()
+                    .subscribe(this::saveContributionsToDB, error -> {
+                        Timber.e("Failed to fetch contributions: %s", error.getMessage());
+                    }));
+        }
     }
 
-    @Override
-    public void onLoadFinished(@NonNull Loader<Cursor> loader, Cursor cursor) {
+    private void showContributions(List<Contribution> contributions) {
         view.showProgress(false);
-        if (null != cursor && cursor.getCount() > 0) {
+        if(!(contributions==null && contributions.size()==0)){
```

Comment:
if contributions ever was null this code guarantees an NPE. "If contributions _is_ null *and* is empty"

Also avoid `.size==0` checks, `.isEmpty()` is there to be used.

Reply:
"&&" is a shorthand operator if contributions are ever null, contributions.size() will never run, and so an NPE will not happen for sure